flink-optimizer/src/main/java/org/apache/flink/optimizer/plan/Channel.java

```diff
@@ -134,6 +134,10 @@ public Channel(PlanNode sourceNode, TempMode tempMode) {
 	public PlanNode getSource() {
 		return this.source;
 	}
+
+	public void setSource(PlanNode source) {
```

Comment:
Fair enough. Adding the `setSource()` method seems to be less invasive compared to adding several update channel methods for different `PlanNode`s.